File: pages/bing_page.py
from fixtures.test_data import SITES
from pages.base_page import BasePage


class BingPage(BasePage):
    def open(self) -> None:
        self.goto(SITES["bing"]["url"])
        self.wait_for_stable_layout(1000)

    def dismiss_cookie_banner(self) -> None:
        accept = self.page.locator('#bnp_btn_accept, button[id*="accept"]')
        if accept.is_visible(timeout=3000):
            accept.click()
            self.page.wait_for_timeout(500)

    @property
    def search_box(self):
        return self.page.locator('#sb_form_q, input[name="q"]')

    @property
    def logo(self):
        return self.page.locator(
            '#b_logo, .b_logo, a[aria-label="Bing"], header a[href*="bing"]'
        ).first

File: pages/bing.page.ts
import { BasePage } from './base.page';
import { SITES } from '../fixtures/test-data';

export class BingPage extends BasePage {
  async open() {
    await this.goto(SITES.bing.url);
    await this.waitForStableLayout(2000);
  }

  async dismissCookieBanner() {
    const acceptButton = this.page.locator('#bnp_btn_accept, button[id*="accept"]');
    if (await acceptButton.isVisible({ timeout: 3000 }).catch(() => false)) {
      await acceptButton.click();
      await this.page.waitForTimeout(500);
    }
  }

  get searchBox() {
    return this.page.locator('#sb_form_q, input[name="q"]');
  }

  get logo() {
    return this.page.locator('#b_logo, .b_logo');
  }

  get homepageContent() {
    return this.page.locator('#b_content, body');
  }
}
